analyzer/quality-analyzer/src/main/java/eu/fasten/analyzer/qualityanalyzer/QualityAnalyzerPlugin.java

```diff
@@ -0,0 +1,151 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.fasten.analyzer.qualityanalyzer;
+
+import eu.fasten.analyzer.qualityanalyzer.data.QAConstants;
+
+import eu.fasten.core.plugins.KafkaPlugin;
+import eu.fasten.core.plugins.DBConnector;
+
+import org.json.JSONObject;
+import org.pf4j.Extension;
+import org.pf4j.Plugin;
+import org.pf4j.PluginWrapper;
+
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.List;
+import java.util.Collections;
+
+import org.jooq.DSLContext;
+
+
+public class QualityAnalyzerPlugin extends Plugin {
+
+    public QualityAnalyzerPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class QualityAnalyzer implements KafkaPlugin, DBConnector {
+
+        private final Logger logger = LoggerFactory.getLogger(QualityAnalyzer.class.getName());
+        private String consumerTopic = "fasten.RapidPlugin.out";
+        private static MetadataUtils utils = null;
+        private Throwable pluginError = null;
+
+        @Override
+        public void setDBConnection(Map<String, DSLContext> dslContexts) {
+            QualityAnalyzer.utils = new MetadataUtils(dslContexts);
+        }
+
+        @Override
+        public Optional<List<String>> consumeTopic() {
+            return Optional.of(Collections.singletonList(consumerTopic));
+        }
+
+        @Override
+        public void setTopic(String topicName) {
+            this.consumerTopic = topicName;
+        }
+
+        @Override
+        public void consume(String kafkaMessage) {
+
+            logger.info("Consumed: " + kafkaMessage);
+
+            var jsonRecord = new JSONObject(kafkaMessage);
+
+            String forge = null;
+
+            if (jsonRecord.has("payload")) {
+                forge = jsonRecord
+                        .getJSONObject("payload")
+                        .getString("forge".replaceAll("[\\n\\t ]", ""));
+            }
+
+            logger.info("forge = " + forge);
+
+            this.pluginError = null;
+
+            if (forge != null) {
+                try {
+                    utils.updateMetadataInDB(forge, jsonRecord);
+                } catch(Exception ex) {
+                    setPluginError(ex);
+                }
+            } else {
+                logger.error("Could not extract forge from the message");
```

Comment:
Please here add something like `this.pluginError = new RuntimeError("Could not extract forge from the message");`. Otherwise, when there is no forge, it will skip the record, and it will seem that it was processed successfully (no errors) while in reality, it wasn't successful at all and there was an error (missing forge).

Reply:
OK done, with a little bit refactoring and setting pluginError to RuntimeException().